File: App/src/main/java/it/gruppoinfor/home2work/custom/OngoinShareView.java
package it.gruppoinfor.home2work.custom;

import android.content.Context;
import android.content.Intent;
import android.util.AttributeSet;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.LinearLayout;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.annimon.stream.Stream;

import butterknife.BindView;
import butterknife.ButterKnife;
import it.gruppoinfor.home2work.R;
import it.gruppoinfor.home2work.activities.OngoingShareActivity;
import it.gruppoinfor.home2workapi.model.Guest;
import it.gruppoinfor.home2workapi.model.Share;

import static it.gruppoinfor.home2work.activities.OngoingShareActivity.EXTRA_SHARE;

public class OngoinShareView extends FrameLayout {

    @BindView(R.id.text_ongoing_share_info)
    TextView textOngoingShareInfo;
    @BindView(R.id.text_ongoin_share_guests)
    TextView textOngoinShareGuests;
    @BindView(R.id.share_item_ongoing_container)
    RelativeLayout shareItemOngoingContainer;
    @BindView(R.id.layout_guests_number)
    LinearLayout layoutGuestsNumber;
    private Context mContext;

    public OngoinShareView(Context context) {
        super(context);
        mContext = context;
        initUI();
    }

    public OngoinShareView(Context context, AttributeSet attributes) {
        super(context, attributes);
        mContext = context;
        initUI();
    }

    public OngoinShareView(Context context, AttributeSet attributeSet, int defStyleAttr) {
        super(context, attributeSet, defStyleAttr);
        mContext = context;
        initUI();
    }

    @Override
    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
        super.onLayout(changed, left, top, right, bottom);
    }

    private void initUI() {
        View view = inflate(getContext(), R.layout.layout_share_ongoing, this);
        ButterKnife.bind(this, view);
    }

    public void setShare(Share share) {

        long guestSize = Stream.of(share.getGuests()).filter(value -> !value.getStatus().equals(Guest.Status.CANCELED)).count();
        textOngoinShareGuests.setText(String.valueOf(guestSize));

        if (share.getType() == Share.Type.DRIVER) {

            textOngoingShareInfo.setText(R.string.layout_ongoing_share_host);

        } else {

            textOngoingShareInfo.setText(R.string.layout_ongoing_share_guest);
            layoutGuestsNumber.setVisibility(GONE);
        }

        shareItemOngoingContainer.setOnClickListener(view -> {
            Intent intent = new Intent(mContext, OngoingShareActivity.class);
            intent.putExtra(EXTRA_SHARE, share);
            mContext.startActivity(intent);
        });

    }
}

File: App/src/main/java/it/gruppoinfor/home2work/custom/OngoinShareView.kt
package it.gruppoinfor.home2work.custom

import android.content.Context
import android.graphics.Rect
import android.util.AttributeSet
import android.view.View
import android.widget.FrameLayout
import it.gruppoinfor.home2work.R
import it.gruppoinfor.home2workapi.model.Guest
import it.gruppoinfor.home2workapi.model.Share
import kotlinx.android.synthetic.main.layout_share_ongoing.view.*

class OngoinShareView : FrameLayout {

    constructor(context: Context) : super(context) {

        View.inflate(context, R.layout.layout_share_ongoing, this)

    }

    constructor(context: Context, attributes: AttributeSet) : super(context, attributes) {

        View.inflate(context, R.layout.layout_share_ongoing, this)

    }

    constructor(context: Context, attributeSet: AttributeSet, defStyleAttr: Int) : super(context, attributeSet, defStyleAttr) {

        View.inflate(context, R.layout.layout_share_ongoing, this)

    }

    fun setShare(share: Share?) {

        val guestSize = share?.guests?.count { guest -> guest.status != Guest.Status.CANCELED }

        text_ongoin_share_guests.text = guestSize.toString()

        if (share?.type == Share.Type.DRIVER) {

            text_ongoing_share_info.setText(R.string.layout_ongoing_share_host)

        } else {

            text_ongoing_share_info.setText(R.string.layout_ongoing_share_guest)
            layout_guests_number.visibility = View.GONE
        }

    }

    override fun onFocusChanged(gainFocus: Boolean, direction: Int, previouslyFocusedRect: Rect?) {
        super.performClick()
        super.onFocusChanged(gainFocus, direction, previouslyFocusedRect)
    }
}
